repositories/gcs.go

```diff
@@ -117,21 +108,56 @@ func (gcs *GCSRepo) DownloadRelease(version, destDir, destFile string) (string,
 	return httputil.DownloadBinary(url, destDir, destFile)
 }
 
+func (gcs *GCSRepo) removeCandidates(history []string, lastN int) ([]string, error) {
+	var resolvedLimit int
+	if lastN < 1 {
+		resolvedLimit = len(history)
+	} else {
+		resolvedLimit = lastN
+	}
+
+	hpos := len(history) - 1
+	descendingReleases := make([]string, 0)
+	for hpos >= 0 && len(descendingReleases) < resolvedLimit {
```

Comment:
Could we use a for-loop with all three parameters here, something like:

```
for hpos := len(history) -1; hpos >= 0 && len(descendingReleases) < resolvedLimit; hpos-- {
  ...
}
```

?